### PR TITLE
REL-3193: Support Subaru Intensive Programs in PIT

### PR DESCRIPTION
**Pull request by cquiroz**
This PR adds a new proposal class to the PIT: `Subaru Intensive Proposal` class. The PR updates the model, adds UI controls to select the proposal properties and includes a new backend for submissions.

Here's a screenshot

screen shot 2017-07-14 at 12 23 41 pm

### CODE REVIEW

File: bundle/edu.gemini.model.p1/src/main/xsd/ProposalClass.xsd [hunk 1]
```diff
@@ -129,6 +130,21 @@
...
+        Subaru Intensive Program
+    -->
+    <xsd:complexType name="SubaruIntensiveProgramClass">
+        <xsd:complexContent>
+            <xsd:extension base="ProposalClass">
+                <xsd:sequence>
+                    <xsd:element name="telescope" type="ExchangeTelescope"/>
+                    <xsd:element name="submission" type="SubaruIntensiveProgramSubmission"/>
+                </xsd:sequence>
+                <xsd:attribute name="tooOption" type="TooOption" use="optional"/>
```
Comment (swalker2m) [inline bundle/edu.gemini.model.p1/src/main/xsd/ProposalClass.xsd]: Subaru supports ToO?

File: bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/view/partner/PartnerView.scala [hunk 2]
```diff
@@ -267,26 +310,30 @@ class PartnerView extends BorderPanel with BoundView[Proposal] {view =>
...
-          case l: LargeProgramClass          =>
+          case l: LargeProgramClass           =>
             selection.item = l.tooOption
             visible = true
-          case f: FastTurnaroundProgramClass =>
+          case f: FastTurnaroundProgramClass  =>
             selection.item = f.tooOption
             visible = true
-          case _                             =>
+          case SubaruIntensiveProgramClass(_, _, _, ExchangeTelescope.GEMINI, tooOption, _) =>
```
Comment (swalker2m) [inline bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/view/partner/PartnerView.scala]: What's the `ExchangeTelescope` and why is it `GEMINI`?

File: bundle/edu.gemini.model.p1/src/main/xsd/ProposalClass.xsd (hunk 1 above)
Comment (cquiroz) [inline bundle/edu.gemini.model.p1/src/main/xsd/ProposalClass.xsd]: No, for this type of proposal you have to select a telescope, which I call an `ExchangeTelescope` for lack of a better word. If your telescope is `Gemini` you have the ToO choice

File: bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/view/partner/PartnerView.scala (hunk 2 above)
Comment (cquiroz) [inline bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/view/partner/PartnerView.scala]: I probably answered this above

**Comment by cquiroz** [comment]
There is no work on ITAC for this. we will not import them into ITAC like FT

**Comment by cquiroz** [comment]
BTW, I want to get this in to produce a prototype, lt is likely that there will be changes after user testing

File: bundle/edu.gemini.model.p1/src/main/xsd/ProposalClass.xsd (hunk 1 above)
Comment (swalker2m) [inline bundle/edu.gemini.model.p1/src/main/xsd/ProposalClass.xsd]: So it is "subaru intensive" but has some Gemini observations?

Comment (cquiroz) [inline bundle/edu.gemini.model.p1/src/main/xsd/ProposalClass.xsd]: The requirement is that for this type of proposal you select the observatory and that defines the selection of the instruments either Gemini or Subaru

**Comment by cquiroz** [comment]
You have a point on questioning the impact on phase II, this should be considered. I'll post a note on the jira ticket

File: bundle/edu.gemini.model.p1/src/main/xsd/ProposalClass.xsd (hunk 1 above)
Comment (swalker2m) [inline bundle/edu.gemini.model.p1/src/main/xsd/ProposalClass.xsd]: I read the information in REL-3193 but I'm still not sure what this is all about.  Are these LP programs for Gemini or are the observations at Subaru or maybe both?

**Comment by cquiroz** [comment]
Updated according to the latest specifications

**Pull request merged**